### DISCUSSION TITLE
Semester 4 Week 4

### DISCUSSION THREAD

**Discussion by Stev3eM** [SE-TINF22B6/CookHub#173] [Show and tell]
Hello backpackers,

This is our weekly discussion-blog to keep you up to date within our project progression.
Another week, another update for you:

<br>

## Project Progress

✅ We updated the prompt for ChatGPT to adventurize recipes to [version 1.0](https://github.com/SE-TINF22B6/CookHub/wiki/adventurize-it-%E2%92%B8-%5B-v1.0-%5D)

<img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/c45f9a25-64f5-4136-9625-ae68c4682aaa" width="400" />

<br><br>

✅ Created recipe base data for the CookHub recipe catalog

<br>

✅ Fixed some bugs 
- saving database entities 
- improved serialization for API endpoint results

<br><br>

## Demo / Testing

✅ Discussed about which testings to implement

__Testing Tools:__ GitHub Actions, Selenium (E2E), xUnit.NET

| Frontend | Backend |
|:--|:--|
| Unit tests for frontend logic | Integration tests |
| E2E tests for Key Features | API tests |

✅ Started planning what and how to demo show.
- Web Surface and Usage
- Login function
- API (Adventurize function, how we use OpenAI)
- Database structure and implementation
- Carlos
- Showcase how a user can create and adventurize a recipe

<br><br>

## Team Building

✅ Created cool memes:

| <img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/4786d86f-070a-422b-ba71-8cc3de59220a" width="300"/> | <img src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/58a7ce25-5a81-4004-a432-04badf0793d3" width="400"/> |
|---|---|

<br><br>

See you next time!

**Your**   <img width="50" alt="image" src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/1a218c37-d3b0-4c88-9ae1-827b5a4eba42">   **Team**

**Comment by giomskii** [comment]
Hello Team CookHub,
your choice of testing tools is quite intriguing, have you already implemented some tests or is this just your current plan?
I also appreciate your team building efforts, very important! 👍🏼 
Keep up the good work - Team DHBWhub

**Reply by Stev3eM** [reply]
We've just planned to implement the tests except of the integrated GitHub Action tests yet. 🍤

**Comment by NicolasVien** [comment]
Great job planning and organizing! Your dedication to quality and functionality is evident in your decision on testing tools and comprehensive demo plan. It's also great to see team-building activities incorporated into your work. Keep up the excellent work towards delivering a high-quality product!

**Comment by BenjaminB109** [comment]
Dear CookHub Team,
good work with the tests. I'm a fan of your team building. Keep up the good work.

**Comment by sfrancisw** [comment]
Your progress this week is impressive! It's clear that you're actively refining and advancing your project. 
Your approach to testing is thorough, with a thoughtful selection of tools and strategies for both frontend and backend. 
Keep up the fantastic work, and I'm excited to see your continued progress!
Your team time2shine
